Add blogpost for recent Supervisor changes (#698)
home-assistant/developers.home-assistant: blog/2020-10-23-supervisor-update.md

---
author: Joakim Sørensen
authorURL: https://github.com/ludeeus
authorTwitter: ludeeus
title: "Calling all add-on developers!"
---

## Documentation

> First off, our [API documentation][api_docs] has been moved to the developer documentation site, during this move it also got a style update to make it easier to navigate. Some of the endpoints are still missing some content, if you have not yet met your quota for [Hacktoberfest] maybe you want to contribute some more details to our API descriptions?

Reviewer: ```suggestion
First off, our [API documentation][api_docs] has been moved to the developer documentation site. During this move it also got a style update to make it easier to navigate. Some of the endpoints are still missing some content. If you have not yet met your quota for [Hacktoberfest] maybe you want to contribute some more details to our API descriptions?
```

This is already in use in our [hassio-addons repository][addons], you can see an example on how we implemented it [here][builder-action].

> It can be used both to ensure that the add-on will still build with changes made to your repository, and to publish the images as part of a release workflow. How to use the action is documented in the [builder repository][builder-action].

Reviewer 2: ```suggestion
It can be used to ensure that the add-on will still build with changes made to your repository and publish the images as part of a release workflow. How to use the action is documented in the [builder repository][builder-action].
```

> Here is an example on how you can use it:

Reviewer 2: ```suggestion
Here is an example of how you can use it:
```

- Deleting snapshots with `POST` calling `/supervisor/snapshots/<slug>/remove` is deprecated, and will be removed later this year. You need to use the `DELETE` method when calling `/supervisor/snapshots/<slug>` instead.
- Using `X-Hassio-Key` header as authentication method is deprecated, and will be removed later this year. You need use an authorization header with a `Bearer` token instead.

The [API documentation][api_docs] have been updated to reflect these changes.

## Add-on options

> The permissions of the `/data/options.json` file, was change from `644` to `600`. If your add-on is running as non-root and you are reading this file, it will give now you permission issues.

Reviewer: ```suggestion
The permissions of the `/data/options.json` file, were changed from `644` to `600`. If your add-on is running as non-root and you are reading this file, it will now give you permission issues.
```

      uses: actions/checkout@v2
    - name: Test build
      uses: home-assistant/builder@master
      with:
        args: |
          --test \
          --all \
          --target /data
```

This example will run a test build on all supported architectures of the add-on.

:::tip
Your repository is mapped to `/data` in the action, so if you have your add-on files in subdirectories, you need to supply `--target /data/{directoryname}` as an argument to the builder action.
:::

## Documentation

> First off, our [API documentation][api_docs] has been moved to the developer documentation site. During this move it also got a style update to make it easier to navigate. Some of the endpoints are still missing some content. If you have not yet met your quota for [Hacktoberfest] maybe you want to contribute some more details to our API descriptions?

Reviewer 2: ```suggestion
First off, our [API documentation][api_docs] has moved to the developer documentation site. During this move, it also got a style update to make it easier to navigate. Some of the endpoints are still missing some content. If you have not yet met your quota for [Hacktoberfest], maybe you want to contribute some more details to our API descriptions?
```

> - If you are using [S6-overlay] in your add-on, you can use [`/etc/fix-attrs.d`][S6-overlay-permissions] to ensure that the user you are running the add-on as, have access to the file.

Reviewer: ```suggestion
- If you are using [S6-overlay] in your add-on, you can use [`/etc/fix-attrs.d`][S6-overlay-permissions] to ensure that the user you are running the add-on as, has access to the file.
```

> - You can change your add-on to run as `root` (default)

Reviewer: ```suggestion
- You can change your add-on to run as `root` (default).
```

> Until now, the Supervisor, our plugins and add-ons have been using a mix of the build number and [Semantic Versioning (SemVer)][semver] as the versioning system, we have decided to replace that for these repositories and to adopt [Calendar Versioning (CalVer)][calver] as our versioning system instead.

Reviewer: ```suggestion
Until now, the Supervisor, our plugins and add-ons have been using a mix of the build number and [Semantic Versioning (SemVer)][semver] as the versioning system. We have decided to replace that for these repositories and to adopt [Calendar Versioning (CalVer)][calver] as our versioning system instead.
```

## API Changes

> - Using the `/homeassistant/*` endpoints is deprecated, and will be removed later this year. You need to use `/core/*` instead.
> - Using `http://hassio/` is deprecated, and will be removed later this year. You need to use `http://supervisor/` instead.
> - Using `HASSIO_TOKEN` is deprecated, and will be removed later this year. You need to use `SUPERVISOR_TOKEN` instead.
> - Deleting snapshots with `POST` calling `/supervisor/snapshots/<slug>/remove` is deprecated, and will be removed later this year. You need to use the `DELETE` method when calling `/supervisor/snapshots/<slug>` instead.
> - Using `X-Hassio-Key` header as authentication method is deprecated, and will be removed later this year. You need use an authorization header with a `Bearer` token instead.

Reviewer 2: ```suggestion
- Using the `/homeassistant/*` endpoints is deprecated and will be removed later this year. You need to use `/core/*` instead.
- Using `http://hassio/` is deprecated and will be removed later this year. You need to use `http://supervisor/` instead.
- Using `HASSIO_TOKEN` is deprecated and will be removed later this year. You need to use `SUPERVISOR_TOKEN` instead.
- Deleting snapshots with `POST` calling `/supervisor/snapshots/<slug>/remove` is deprecated and will be removed later this year. You need to use the `DELETE` method when calling `/supervisor/snapshots/<slug>` instead.
- Using `X-Hassio-Key` header as an authentication method is deprecated and will be removed later this year. You need to use an authorization header with a `Bearer` token instead.
```

> The [API documentation][api_docs] have been updated to reflect these changes.

Reviewer 2: ```suggestion
The [API documentation][api_docs] has been updated to reflect these changes.
```

## Add-on options

> The permissions of the `/data/options.json` file, were changed from `644` to `600`. If your add-on is running as non-root and you are reading this file, it will now give you permission issues.

Reviewer 2: ```suggestion
The permissions of the `/data/options.json` file, is changed from `644` to `600`. If your add-on is running as non-root and reading this file, it will now give you permission issues.
```

> We are migrating the Supervisor from release based development to continuous development. This fits perfectly with our exists channel based update strategy (stable, beta and dev). We are now leveraging automated pipelines, to build and push out new Supervisor versions to the correct channels. By moving to this structure there was no more need to a dual branch setup, so both our `dev` and `master` branches have now been replaced with a new `main` branch. Our plugins (DNS, Multicast, Observer, CLI) for the Supervisor will also follow this continuous development principal.

Reviewer: ```suggestion
We are migrating the Supervisor from release based development to continuous development. This fits perfectly with our existing channel based update strategy (stable, beta and dev). We are now leveraging automated pipelines, to build and push out new Supervisor versions to the correct channels. By moving to this structure there was no more need for a dual branch setup, so both our `dev` and `master` branches have now been replaced with a new `main` branch. Our plugins (DNS, Multicast, Observer, CLI) for the Supervisor will also follow this continuous development principle.
```

> We did this move to provide higher software quality with automatic test system. Every commit now trigger a new dev release which get tested by our test instances and issues are imminently reported to sentry, this gives us the opportunity to test all changes before we create a releases, when a release is created the the changes are available to the beta channel. Based on the feedback and reports to Sentry we can decide to push this into stable channel.

Reviewer: ```suggestion
We did this move to provide higher software quality with an automatic test system. Every commit now triggers a new dev release which get tested by our test instances and issues are immediately reported to Sentry. This gives us the opportunity to test all changes before we create a release. When a release is created the changes are made available to the beta channel. Based on the feedback and reports to Sentry we can decide to push this into the stable channel.
```

> We are using [Azure pipelines][azure_pipelines] to build and publish the Supervisor. If you are interested on how we are doing this, you can look at the [release pipeline here][release_pipeline], and the [template repository here][azure_templates].

Reviewer: ```suggestion
We are using [Azure pipelines][azure_pipelines] to build and publish the Supervisor. If you are interested in how we are doing this, you can look at the [release pipeline here][release_pipeline], and the [template repository here][azure_templates].
```

There are several steps you can do in your add-on to continue to use this information:

- If you are using [S6-overlay] in your add-on, you can use [`/etc/fix-attrs.d`][S6-overlay-permissions] to ensure that the user you are running the add-on as, has access to the file.
- You can change your add-on to run as `root` (default).

## Releases

Until now, the Supervisor, our plugins and add-ons have been using a mix of the build number and [Semantic Versioning (SemVer)][semver] as the versioning system. We have decided to replace that for these repositories and to adopt [Calendar Versioning (CalVer)][calver] as our versioning system instead.

> We are migrating the Supervisor from release based development to continuous development. This fits perfectly with our existing channel based update strategy (stable, beta and dev). We are now leveraging automated pipelines, to build and push out new Supervisor versions to the correct channels. By moving to this structure there was no more need for a dual branch setup, so both our `dev` and `master` branches have now been replaced with a new `main` branch. Our plugins (DNS, Multicast, Observer, CLI) for the Supervisor will also follow this continuous development principle.

Reviewer 2: ```suggestion
We are migrating the Supervisor from release based development to continuous development. This fits perfectly with our existing channel-based update strategy (stable, beta and dev). We are now leveraging automated pipelines to build and push out new Supervisor versions to the correct channels. There was no more need for a dual branch setup by moving to this structure, so both our `dev` and `master` branches have now been replaced with a new `main` branch. Our plugins (DNS, Multicast, Observer, CLI) for the Supervisor will also follow this continuous development principle.
```

> It can be used both to ensuring that the add-on will still build with changes made to your repository, and to publish the images as a part of a release workflow. How to use the action is documented in the [builder repository][builder-action].

Reviewer: ```suggestion
It can be used both to ensure that the add-on will still build with changes made to your repository, and to publish the images as part of a release workflow. How to use the action is documented in the [builder repository][builder-action].
```

> We did this move to provide higher software quality with automatic test system. Every commit now trigger a new dev release which get tested by our test instances. Issues are imminently reported to sentry. This gives us the opportunity to test all changes before we create a releases. When a release is created the changes will come available in the beta channel. Once declared stable, we can promote the release to the stable channel.

Reviewer: ```suggestion
We did this move to provide higher software quality with an automatic test system. Every commit now triggers a new dev release which get tested by our test instances. Issues are imminently reported to sentry. This gives us the opportunity to test all changes before we create a release. When a release is created the changes will come available in the beta channel. Once declared stable, we can promote the release to the stable channel.
```

> We did this move to provide higher software quality with an automatic test system. Every commit now triggers a new dev release which get tested by our test instances. Issues are imminently reported to sentry. This gives us the opportunity to test all changes before we create a release. When a release is created the changes will come available in the beta channel. Once declared stable, we can promote the release to the stable channel.

Reviewer 2: ```suggestion
We made this move to provide higher software quality with an automatic test system. Every commit now triggers a new dev release, which gets tested by our test instances. Issues are imminently reported to sentry. This gives us the opportunity to test all changes before we create a release. When a release is created, the changes will come available in the beta channel. Once declared stable, we can promote the release to the stable channel.
```

> We are using [Azure pipelines][azure_pipelines] to build and publish the Supervisor. If you are interested in how we are doing this, you can look at the [release pipeline here][release_pipeline], and the [template repository here][azure_templates].

Author: This is no longer true and needs to be rewritten

---
author: Joakim Sørensen
authorURL: https://github.com/ludeeus
authorTwitter: ludeeus

> title: "Calling all add-on developers!"

Reviewer 3: Let's make the title more descriptive.

```suggestion
title: "Upcoming changes to add-ons"
```

---

## Documentation

First off, our [API documentation][api_docs] has been moved to the developer documentation site, during this move it also got a style update to make it easier to navigate. Some of the endpoints are still missing some content, if you have not yet met your quota for [Hacktoberfest] maybe you want to contribute some more details to our API descriptions?

## API Changes

- Using the `/homeassistant/*` endpoints is deprecated, and will be removed later this year. You need to use `/core/*` instead.
- Using `http://hassio/` is deprecated, and will be removed later this year. You need to use `http://supervisor/` instead.
- Using `HASSIO_TOKEN` is deprecated, and will be removed later this year. You need to use `SUPERVISOR_TOKEN` instead.
- Deleting snapshots with `POST` calling `/supervisor/snapshots/<slug>/remove` is deprecated, and will be removed later this year. You need to use the `DELETE` method when calling `/supervisor/snapshots/<slug>` instead.
- Using `X-Hassio-Key` header as authentication method is deprecated, and will be removed later this year. You need use an authorization header with a `Bearer` token instead.

> The [API documentation][api_docs] have been updated to reflect these changes.

Reviewer 3: Looking at the docs, I don't see the old APIs. Shouldn't they be there but marked as deprecated? Or are we going for just the current APIs?

Author: If we document old deprecated endpoints, folks will use them.
I think the docs should reflect the current non-deprecated endpoints and methods.

Many of these were "soft" deprecated earlier, but I could not find anywhere that we had communicated that to add-on devs

- If you are using [S6-overlay] in your add-on, you can use [`/etc/fix-attrs.d`][S6-overlay-permissions] to ensure that the user you are running the add-on as, have access to the file.
- You can change your add-on to run as `root` (default)

## Releases

Until now, the Supervisor, our plugins and add-ons have been using a mix of the build number and [Semantic Versioning (SemVer)][semver] as the versioning system, we have decided to replace that for these repositories and to adopt [Calendar Versioning (CalVer)][calver] as our versioning system instead.

We are migrating the Supervisor from release based development to continuous development. This fits perfectly with our exists channel based update strategy (stable, beta and dev). We are now leveraging automated pipelines, to build and push out new Supervisor versions to the correct channels. By moving to this structure there was no more need to a dual branch setup, so both our `dev` and `master` branches have now been replaced with a new `main` branch. Our plugins (DNS, Multicast, Observer, CLI) for the Supervisor will also follow this continuous development principal.

> We did this move to provide higher software quality with automatic test system. Every commit now trigger a new dev release which get tested by our test instances and issues are imminently reported to sentry, this gives us the opportunity to test all changes before we create a releases, when a release is created the the changes are available to the beta channel. Based on the feedback and reports to Sentry we can decide to push this into stable channel.

Reviewer 3: ```suggestion
We did this move to provide higher software quality with automatic test system. Every commit now trigger a new dev release which get tested by our test instances. Issues are imminently reported to sentry. This gives us the opportunity to test all changes before we create a releases. When a release is created the changes will come available in the beta channel. Once declared stable, we can promote the release to the stable channel.
```

We are using [Azure pipelines][azure_pipelines] to build and publish the Supervisor. If you are interested on how we are doing this, you can look at the [release pipeline here][release_pipeline], and the [template repository here][azure_templates].

## Private Container Registries

> In the [Supervisor 249][private-container-registries] blog post, we mentioned the introduction of private container registries. For you as a add-on maintainer this now give you the possibility to use any container registries like GitHub's package registry for docker, or your own instance of a container registry, while developing or testing your add-on.

Reviewer 3: ```suggestion
In the [Supervisor 249][private-container-registries] blog post, we mentioned the introduction of private container registries. Add-on maintainers now have the possibility to use any container registries, like GitHub's package registry for docker, or your own instance of a container registry.
```

This can also be used for private testing of your add-on by giving pull access to users.

To make the Supervisor use your credentials, you can use the [API][docker_api], the [CLI][cli], or the Supervisor panel in the Home Assistant UI.

> ## GitHub Action

Reviewer 3: Should this be higher in the post? Seems like the most important thing.

Author: It's the most exciting for sure, but most important :man_shrugging: 
Moved it to the top now.

> You can now use our [builder][hassio-builder] as a [GitHub action][github_action]! :tada:

Reviewer 3: ```suggestion
You can now use our [add-on builder][hassio-builder] as a [GitHub action][github_action]! :tada:
```

Author: Not sure about this, the targeted audience is add-on devs, but the builder is for more

Reviewer 3: Right, can we then add the things it can build ?

Author: There is really no limits on what it can build, but we can add what we use it for.
